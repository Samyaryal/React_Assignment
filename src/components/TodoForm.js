import React from 'react';
import { Button, Form } from 'react-bootstrap';

const TodoForm = ({ onSubmit, value, onChange }) => {
  return (

    <Form onSubmit={onSubmit} data-testid="form">
      <Form.Group>
        <Form.Label><b>Add Todo</b></Form.Label>
        <Form.Control type="text" placeholder="Add a new todo.." value={value} onChange={onChange} required />
      </Form.Group>
      <Button  variant="primary mb-3" type="submit">Add</Button>
    </Form>

  )
}

export default TodoForm;