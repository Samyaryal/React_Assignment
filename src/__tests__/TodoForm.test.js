import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import TodoForm from '../components/TodoForm';

it("submits", () => {
  const onSubmit = jest.fn();
  const { getByTestId} = render(<TodoForm onSubmit={onSubmit} />);
  fireEvent.submit(getByTestId("form"));
  expect(onSubmit).toHaveBeenCalled();
});

//snapshot testing
it('submits ', () => {
  const onSubmit = jest.fn();
  const { getByTestId } = render(<TodoForm  onSubmit={onSubmit} />);
  fireEvent.submit(getByTestId("form"));
  expect(getByTestId).toMatchSnapshot();
});

