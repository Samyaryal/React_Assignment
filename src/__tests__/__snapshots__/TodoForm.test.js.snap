// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it submits  1`] = `[Function]`;

exports[`submits  1`] = `[Function]`;
